src/tests/helpers/test-store-purchase-helper.test.ts

```diff
@@ -0,0 +1,253 @@
+        storeTransactionId: expect.stringMatching(/^test_.*test-uuid-123$/),
+        productIdentifier: "monthly_trial_intro",
+        purchaseDate: expect.any(Date),
+      },
+    });
+  });
+
+  test("rejects with error on failed purchase", async () => {
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    props?.onFailedPurchase();
+
+    await expect(promise).rejects.toThrow(PurchasesError);
+    await expect(promise).rejects.toThrow("Simulated test purchase failure");
+  });
+
+  test("rejects with user cancelled error on cancel", async () => {
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    props?.onCancel();
+
+    await expect(promise).rejects.toThrowError(PurchasesError);
+  });
+
+  test("handles backend error during purchase", async () => {
+    const backendError = new Error("Network error");
+    vi.mocked(mockBackend.postReceipt).mockRejectedValue(backendError);
+
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    props?.onValidPurchase();
+
+    await expect(promise).rejects.toThrow("Network error");
+  });
+
+  test("cleans up DOM and component on successful purchase", async () => {
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    await props?.onValidPurchase();
+    await promise;
+
+    expect(unmount).toHaveBeenCalledWith(mockComponent);
+  });
+
+  test("cleans up DOM and component on failed purchase", async () => {
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    props?.onFailedPurchase();
+
+    try {
+      await promise;
+      expect(false).toBe(true); // Should not reach here
+    } catch {
+      // Expected error
+    }
+
+    expect(unmount).toHaveBeenCalledWith(mockComponent);
+  });
+
+  test("cleans up DOM and component on cancel", async () => {
+    const promise = purchaseTestStoreProduct(
+      mockPurchaseParams,
+      mockBackend,
+      "test-user-id",
+    );
+
+    const mountCall = vi.mocked(mount).mock.calls[0];
+    const props = mountCall[1].props;
+
+    props?.onCancel();
+
+    try {
+      await promise;
+      expect(false).toBe(true); // Should not reach here
+    } catch {
+      // Expected error
+    }
```

Comment:
nit: Using the `expectPromiseToError` helper function will make this a bit cleaner:

```suggestion
    await expectPromiseToError(
      promise,
      new PurchasesError(ErrorCode.UserCancelledError)
    );
```